src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,343 @@
+  ∃ S : set α, measurable_set S ∧
+    j.to_signed_measure ≤[S] 0 ∧ 0 ≤[Sᶜ] j.to_signed_measure ∧ j.μ S = 0 ∧ j.ν Sᶜ = 0 :=
+begin
+  obtain ⟨S, hS₁, hS₂, hS₃⟩ := j.mutually_singular,
+  refine ⟨S, hS₁, _, _, hS₂, hS₃⟩,
+  { refine restrict_le_restrict_of_subset_le _ _ (λ A hA hA₁, _),
+    rw [to_signed_measure, sub_to_signed_measure_apply hA,
+        show j.μ A = 0, by exact nonpos_iff_eq_zero.1 (hS₂ ▸ measure_mono hA₁),
+        ennreal.zero_to_real, zero_sub, neg_le, zero_apply, neg_zero],
+    exact ennreal.to_real_nonneg },
+  { refine restrict_le_restrict_of_subset_le _ _ (λ A hA hA₁, _),
+    rw [to_signed_measure, sub_to_signed_measure_apply hA,
+        show j.ν A = 0, by exact nonpos_iff_eq_zero.1 (hS₃ ▸ measure_mono hA₁),
+        ennreal.zero_to_real, sub_zero],
+    exact ennreal.to_real_nonneg },
+end
+
+end jordan_decomposition
+
+namespace signed_measure
+
+open measure vector_measure jordan_decomposition classical
+
+variables {s : signed_measure α} {μ ν : measure α} [finite_measure μ] [finite_measure ν]
+
+/-- Given a signed measure `s`, `s.to_jordan_decomposition` is the Jordan decomposition `j`,
+such that `s = j.to_signed_measure`. This property is known as the Jordan decomposition
+theorem, and is shown by `signed_measure.to_jordan_decomposition_to_signed_measure_eq`. -/
+def to_jordan_decomposition (s : signed_measure α) : jordan_decomposition α :=
+let i := some s.exists_compl_positive_negative in
+let hi := some_spec s.exists_compl_positive_negative in
+{ μ := s.to_measure_of_zero_le i hi.1 hi.2.1,
+  ν := s.to_measure_of_le_zero iᶜ hi.1.compl hi.2.2,
+  μ_finite := infer_instance,
+  ν_finite := infer_instance,
+  mutually_singular :=
+  begin
+    refine ⟨iᶜ, hi.1.compl, _, _⟩,
+    { rw [to_measure_of_zero_le_apply _ _ hi.1 hi.1.compl], simpa },
+    { rw [to_measure_of_le_zero_apply _ _ hi.1.compl hi.1.compl.compl], simpa }
+  end }
+
+lemma to_jordan_decomposition_spec (s : signed_measure α) :
+  ∃ (i : set α) (hi₁ : measurable_set i) (hi₂ : 0 ≤[i] s) (hi₃ : s ≤[iᶜ] 0),
+    s.to_jordan_decomposition.μ = s.to_measure_of_zero_le i hi₁ hi₂ ∧
+    s.to_jordan_decomposition.ν = s.to_measure_of_le_zero iᶜ hi₁.compl hi₃ :=
+begin
+  set i := some s.exists_compl_positive_negative,
+  obtain ⟨hi₁, hi₂, hi₃⟩ := some_spec s.exists_compl_positive_negative,
+  exact ⟨i, hi₁, hi₂, hi₃, rfl, rfl⟩,
+end
+
+/-- **The Jordan decomposition theorem**: Given a signed measure `s`, there exists a pair of
+mutually singular measures `μ` and `ν` such that `s = μ - ν`. In this case, the measures `μ`
+and `ν` are given by `s.to_jordan_decomposition.μ` and `s.to_jordan_decomposition.ν` respectively.
+
+Note that we use `jordan_decomposition.to_signed_measure` to represent the signed measure
+corresponding to `s.to_jordan_decomposition.μ - s.to_jordan_decomposition.ν`. -/
+@[simp] lemma to_jordan_decomposition_to_signed_measure_eq (s : signed_measure α) :
+  s.to_jordan_decomposition.to_signed_measure = s :=
```

Comment:
```suggestion
@[simp] lemma to_jordan_decomposition_to_signed_measure (s : signed_measure α) :
  s.to_jordan_decomposition.to_signed_measure = s :=
```

Reply:
Would the name also change to `to_signed_measure_to_jordan_decomposition`?